test/unit/streams.test.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import util from 'util';
+import Stream, { PassThrough } from 'stream';
+import { gzipSync } from 'zlib';
+
+import {
+  CheapJsonChecker,
+} from '../../src/utils/streams';
+
+// In this test file, we're using the pipeline function, mainly because that's
+// how we'll use these streams in the app code.
+const pipeline = util.promisify(Stream.pipeline);
+const nextTick = util.promisify(process.nextTick);
+describe('EarlyContentChecker', () => {
+  it('accepts normal content', async () => {
+    const fixture = '{ "foo": "bar" }';
+    const checker = new CheapJsonChecker();
+    const input = new PassThrough();
+    input.write(fixture.slice(0, 3));
+    input.end(fixture.slice(3));
+    await expect(pipeline(input, checker)).resolves.toBe(undefined);
+  });
+
+  it('accepts content with some space at the start', async () => {
```

Comment:
Does this actually happen with our profiles?

Reply:
Not when feeding directly from the UI but I'm also thinking of some more "manual" use cases where profiles could be posted (for example from CI). Maybe too much precaution from my side...